content/en/docs/concepts/security/mutating-webhooks-good-practices.md

```diff
@@ -0,0 +1,585 @@
+  namespace.
+* Limit access to edit the webhook configuration resources. 
+* Limit access to the namespace that runs the webhook server, if the server is
+  in-cluster.
+* Consider cluster availability needs during webhook design.
+
+### Install and enable a mutating webhook {#install-enable-mutating-webhook}
+
+When you're ready to deploy your mutating webhook to a cluster, use the
+following order of operations: 
+
+1.  Install the webhook server and start it.
+1.  Set the `failurePolicy` field in the `MutatingWebhookConfiguration` object
+    to Ignore. This lets you avoid disruptions caused by misconfigured webhooks.
+1.  Set the `namespaceSelector` field in the `MutatingWebhookConfiguration`
+    object to a test namespace.
+1.  Deploy the `MutatingWebhookConfiguration` object to your cluster.
+
+Monitor the webhook in the test namespace to check for any issues, then roll the
+webhook out to other namespaces. If the webhook intercepts an API request that
+it wasn't meant to intercept, pause the rollout and adjust the scope of the
+webhook configuration.
+
+### Limit edit access to mutating webhooks {#limit-edit-access}
+
+Mutating webhooks are powerful Kubernetes controllers. Use RBAC or another
+authorization mechanism to limit access to your webhook configurations and
+servers. Ensure that the following access is only available to trusted
+entities:
+
+* Verbs: `create`, `update`, `patch`, `delete`, `deletecollection`
+* API group: `admissionregistration.k8s.io/v1`
+* Resources: `MutatingWebhookConfigurations`
+
+If your mutating webhook server runs in the cluster, limit access to create or
+modify any resources in that namespace.
+
+### Use a high-availability deployment model {#ha-deployment}
+
+Consider your cluster's availability requirements when designing your webhook. 
+For example, during node downtime or zonal outages, Kubernetes marks Pods as
+`NotReady` to allow load balancers to reroute traffic to available zones and
+nodes. These updates to Pods might trigger your mutating webhooks. Depending on
+the number of affected Pods, the mutating webhook server has a risk of timing
+out or causing delays in Pod processing. As a result, traffic won't get
+rerouted as quickly as you need.
+
+Consider situations like the preceding example when writing your webhooks.
+Exclude operations that are a result of Kubernetes responding to unavoidable
+incidents.
+
+## Examples of good implementations {#example-good-implementations}
+
+{{% thirdparty-content %}}
+
+The following projects are examples of "good" custom webhook server
+implementations:
+
+* [`cert-manager`](https://github.com/cert-manager/cert-manager/tree/master/internal/webhook)
+* [Gatekeeper Open Policy Agent (OPA)](https://open-policy-agent.github.io/gatekeeper/website/docs/mutation)
+
+## {{% heading "whatsnext" %}}
+
+* To learn more about RBAC, see the [RBAC documentation](/docs/reference/access-authn-authz/rbac/).
```

Comment:
Let's also tell people about webhooks for authz: https://k8s.io/docs/reference/access-authn-authz/webhook/

Maybe also mention declarative admission control (ValidatingAdmissionPolicy & MutatingAdmissionPolicy).

Reply:
Ah I havent actually updated this section. The RBAC link was from me copying the RBAC good practices markdown file. I'll leave this comment open as a reminder.